starlite/routes/websocket.py

```diff
@@ -56,10 +57,15 @@ async def handle(self, scope: "WebSocketScope", receive: "Receive", send: "Send"
         if self.route_handler.resolve_guards():
             await self.route_handler.authorize_connection(connection=websocket)
 
-        kwargs = await self._resolve_kwargs(websocket=websocket)
-        await self.route_handler.fn.value(**kwargs)
+        kwargs, cleanup_group = await self._resolve_kwargs(websocket=websocket)
 
-    async def _resolve_kwargs(self, websocket: "WebSocket[Any, Any]") -> Dict[str, Any]:
+        async with cleanup_group:
+            await self.route_handler.fn.value(**kwargs)
+        await cleanup_group.cleanup()
```

Comment:
if youre using this as a context manager - you can call cleanup on the exit from the context.

Reply:
See comment below. The points at which exceptions are thrown and cleanup is called have to be separated for the http routes, so cleanup cannot automatically happen during exit.